scripts/build_docs.py

```diff
@@ -0,0 +1,130 @@
+from typing import Dict
+import os
+from pathlib import Path
+from subprocess import check_output
+import argparse
+
+from ruamel.yaml import YAML
+
+
+exclude_files = [".DS_Store", "__init__.py", "__init__.pyc", "README.md", "version.py", "run.py"]
+
+
+def render_docs(relative_src_path: str, src_file: str, to_file: str, modifier="++") -> None:
+    """
+    Shells out to pydocmd, which creates a .md file from the docstrings of python functions and classes in
+    the file we specify.
+    """
+    relative_src_namespace = relative_src_path.replace("/", ".")
+    src_base = src_file.replace(".py", "")
+
+    if relative_src_namespace == "":
+        namespace = f"allennlp.{src_base}{modifier}"
+    else:
+        namespace = f"allennlp.{relative_src_namespace}.{src_base}{modifier}"
+
+    args = ["pydocmd", "simple", namespace]
+    call_result = check_output(args, env=os.environ).decode("utf-8")
+    with open(to_file, "w") as f:
+        f.write(call_result)
+
+    print(f"Built docs for {src_file}: {to_file}")
+
+
+def build_docs_for_file(
+    relative_path: str, file_name: str, docs_dir: str, check: bool = False
+) -> Dict[str, str]:
+    """
+    Build docs for an individual python file. If `check` is passed, we don't generate the docs themselves,
+    but instead we just generate what the corresponding mkdocs YAML would be, so we can compare it to the
+    current version and see if it has changed (which would mean the docs need re-building).
```

Comment:
It's not clear to me what is in the YAML - if I update a docstring without changing the module structure at all, would this trigger a rebuild?

Reply:
No, it was a halfway house solution to retain some checks - but I'll remove this as we don't need it.